### PR TITLE
Shadowlings 2.0

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
experiment: Shadowlings 2.0
spriteadd: cool sprites by Fury McFlurry
/:cl:

Ported these guys from Hippie. 5x better than gang, kek.
 (thank Fury on hippie for the sprite!)

**Changes from original:**
* Removed drain life
* replaced the jaunt with shadow walk. Slings can go through walls, _if there is darkness on the other side_, and move fast in the dark. _However, they move very slowly in the light, and they are unable to pull things with them through walls_
* Shadowlings can enthrall implanted people, but it takes 65 additional seconds, giving them a chance to escape if they're quick,
* paradise ascendant sprites
* the amount of thralls required to ascend scale with players, (min 15, max 25)
* Shadowlings now use the new method of toggling night vision
* Ascendants can be killed by singulos, now including fluff sounds!
* Glare is no longer "hurr durr i stared at you and now i can cuff you", now it only causes stamina damage and stuttering, amount only based on distance. It still does the same as the old glare when you are _directly next to the sling_, but any further and it's ONLY stamina damage and stuttering.

### CODE REVIEW

**Comment by optimumtact** [comment]
pr does not replace gang

**Comment by MoreRobustThanYou** [comment]
@optimumtact sadly i'd likely get scolded for doing that due to covering too things in one PR

**Comment by ExcessiveUseOfCobblestone** [comment]
@MoreRobustThanYou just make another pr removing gang.

File: code/game/gamemodes/shadowling/shadowling.dm
Comment (lordpidey) [inline code/game/gamemodes/shadowling/shadowling.dm]: We should probably move this if statement to the finalize_shadowling proc, that way we only add an extra timer when we need to, rather than for everyone that is a shadowling.

File: code/game/gamemodes/shadowling/shadowling_items.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_items.dm]: This file should not exist; the shadowling species should have all of the effects of these items.

Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling_items.dm]: i think these are mainly there to prevent them from putting other clothes on. also, this file contains the markings effects

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this must apply proper procs to all possible lit objects, rather than set_light(0), so flashlights turn off properly, lights aren't permanently disabled without a lightswitch, glowshrooms stop glowing but not properly so, etc.

Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: How? also, glowshrooms are killed.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.

File: code/game/gamemodes/shadowling/shadowling_items.dm
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_items.dm]: Species can do both of those.

File: code/modules/mob/mob_defines.dm
Comment (KorPhaeron) [inline code/modules/mob/mob_defines.dm]: yuck

**Review by imtakingabreakdontatme — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
OH [B]YEAH

**Comment by KorPhaeron** [comment]
I hate when people spend lots of time to code and open big PRs that I have said wont be accepted

**Comment by Armhulen** [comment]
IT'S HAPPENINGGGGGGGGGGGGGGGG

(kill gangs so this goes through)


GGGGGGGGGGGGGG

**Comment by Armhulen** [comment]
kor, it's a good conversion mode though because it's simple and easy for thralls while giving complex for people who sign up for sling (the sling's power itself)

File: code/game/gamemodes/shadowling/shadowling_abilities.dm
Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: how would i do that

File: code/modules/mob/mob_defines.dm
Comment (ChangelingRain) [inline code/modules/mob/mob_defines.dm]: yeah no don't do this

**Comment by Armhulen** [comment]
he's removed some problems and there are more to go but doesn't every mode have problems? this just has less

**Comment by Armhulen** [comment]
@MoreRobustThanYou if you want this in as much as i do, remove some of the easy escapes and lower the conversion time a bit to conpensate. 60 seconds should be for implanted people, not for everyone you find.

**Comment by Armhulen** [comment]
@MoreRobustThanYou also amazing sprites i'm proud of you son

**Comment by Armhulen** [comment]
even though they aren't yours lamo!!

**Comment by optimumtact** [comment]
@KorPhaeron this is just a port from hippie so I doubt they had too much work to do

**Comment by Armhulen** [comment]
yeah don't worry, now can we do this??

**Comment by KorPhaeron** [comment]
Actually its a bad conversion mode because

- Its a conversion mode

- It relies on parasting

- It has a very binary win/loss condition where literally everyone dies if you fail to validhunt hard enough, encouraging security to act psychotic and run deathcamps

-Their objective encourages non interaction until they have snowballed enough for it to no longer be fun to fight

-Limited toolset of the shadowlings mean rounds are samey.

**Comment by MoreRobustThanYou** [comment]
Armhulen, i didn't make the sprites, Fury, a hippie sprite did.

**Comment by DoctoreMad** [comment]
>5x better than gang, kek.

0 multiplied by 5 is still 0.

**Comment by MoreRobustThanYou** [comment]
> It has a very binary win/loss condition where literally everyone dies if you fail to validhunt hard enough, encouraging security to act psychotic and run deathcamps

you lose and everybody dies in nukeops if you fail to shoot hard enough, so what?

**Comment by KorPhaeron** [comment]
Yeah it was cool when nuke did it, and then cult/cc/sling all did it too

And at least nuke isnt a conversion mode

Also do you seriously not see the problem with security killing more people than the antags in half our modes and carrying that playstyle into the rest of them?

**Comment by MoreRobustThanYou** [comment]
@KorPhaeron Thralls shouldn't really be killed unless they pose an immediate threat to security, rather they should be detained and dethralled

File: code/modules/mob/mob_defines.dm
Comment (MoreRobustThanYou) [inline code/modules/mob/mob_defines.dm]: How do I do this, then?

**Comment by BeeSting12** [comment]
@MoreRobustThanYou 
Cultists shouldn't really be killed unless they pose an immediate threat to security, rather they should be detained and deculted.

Replace cultist with any conversion antagonist. It won't happen. If security feels they have to kill the thralls then they will just like any other conversion antag

**Comment by GuyonBroadway** [comment]
@BeeSting12 The thing is we come from hippe where the mentality is. "Validboner > everything"

If there was a mindsheild implant in a guys pocket, he had a ganger tied up and no other weapons available he would beat said ganger to death with his fists. 

That being said, I suggested some ideas AFTER we merged slings.

Most notably, "what if the sling ascending is not the endgame" what if the ascendant is strong with potent but not utterly godlike abilities and is also killable. The slings primary objective is to escape on the shuttle while ascended.

**Comment by Super3222** [comment]
> replaced the jaunt with shadow walk. Slings can go through walls, if there is darkness on the other side, and move fast in the dark. However, they move very slowly in the light, and they are unable to pull things with them through walls

Ports from https://github.com/yogstation13/yogstation-old/pull/118 or https://github.com/yogstation13/yogstation/pull/75. Can't tell, or maybe you came up with it on your own.

Also why are you trying to revive this game mode? Wouldn't it be better to complete the spiritual successor, Umbrage, instead. Umbrage was meant to fix all of the problems with shadowlings. The same problems that the design lead is now bringing up because of this PR. I mean to be honest, this barely changes anything in the form of gameplay compared to the previous shadowling mode that got deleted from the codebase. Wouldn't it make sense to work forwards instead of backwards.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Cultists shouldn't really be killed unless they pose an immediate threat to security, rather they should be detained and deculted.

I was under the impression this was because deconv takes 5ever and cultists have a "haha fuck you" teleporting device [at least blood], not because V A L I D Z

**Comment by kilkun** [comment]
this pr gave me an idea

what if nuke WAS a conversion mode :thinking:

**Comment by Durkel** [comment]
Kor is right, every time we add a conversion mode people complain because of various reasons some of which he already stated. How is this going to be any different than  the other times?

**Comment by FuryMcFlurry** [comment]
oh hey i did the pixels, could you add my ckey to the changelog (Fury McFlurry) to the changelog for new changeling hatchling sprite.

**Comment by KorPhaeron** [comment]
@kilkun 

Declare war, wait for suicides, summon them as reinforcements

**Comment by pubby** [comment]
Fucking abort this shit now before it becomes somebody's codebaby.

:baby: :gun:  :gun:  :gun:  :gun:

**Comment by Armhulen** [comment]
@pubby not an argument, it's ported and the excuse could be used for any gamemode

**Comment by Armhulen** [comment]
also it's my codebaby

**Comment by Armhulen** [comment]
grr lets be real mad about shadowling when the problems this has are broad and effect every conversion mode

**Comment by KorPhaeron** [comment]
Yes generally when something bad is happening you seek to stop more of the same instead of doubling down on it and compounding the problem

**Comment by pubby** [comment]
@Armhulen Every hour spent developing/testing/discussing or """improving""" a conversion mode is an hour of time that could be spent improving some other part of the game. With shadowling comes a big steaming pile of debt. We currently have a lot of debt. I mean, there are like 6 conversion modes and all of them are problematic. Add a seventh and there goes the time and effort that could be spent elsewhere.

I want to add a ton of stuff to this game, but more conversion modes are not one of them. Big :-1: from me.

**Comment by kilkun** [comment]
>Yes generally when something bad is happening you seek to stop more of the same instead of doubling down on it and compounding the problem

**Comment by Durkel** [comment]
@pubby Kind of like how the servers been suffering from 30% dilation constantly because something is fucked (probably with the lighting).

 I'd gladly take someone fixing the constant fucking lag then a return of a crappy gamemode.

**Comment by optimumtact** [comment]
To be clear, the maintainers have little to no appetite to merge this, so I wouldn't bother working on it anymore.

It's possible we might accept sling as a secondary antag like revenants.

**Comment by kilkun** [comment]
why not make it a wizard spell

give it the old reverse cult twist

**Comment by KingPhilipIII** [comment]
I play on Paradise frequently, and before anyone else says it, I will preemptively say shut up about Furrystation.

Right. To my point, on Paradise, shadowling is honestly one of my favorite antagonists. It's just fun for me, it may play the same but I still derive immense amounts of enjoyment from it, in part that it's weighted low and I NEVER FUCKING ROLL SHADOWLING FUCK YOU RNGESUS but it works there. The rounds progress at a slower pace and the crew is much more inclined to fight threats than they are to simply flee (In part because admins won't let them call the shuttle without a good excuse.) Thralls are captured, dethralled, busted out of medbay before dethralling, stuff happens and Security is held to a far higher standard of behavior(They can't just set up a death camp) than they are here. Slings tend to win unless Security is on point or the slings just have a bad day and get toasted when they shadow walk into space and burn to death or some shit.

On TG? Jesus Christ no. I love TG and our community, as toxic and terrible as it may be sometimes, but with the player base here I see shadowlings going up in flames. I'm by no means disparaging the mode as I genuinely do enjoy it but I don't see it working out here. Firm thumbs down from me, hate to say it but we need to stop making Sec have a higher body count on average than the antagonists do.

**Comment by optimumtact** [comment]
where did it all go so wrong that paradise supposedly has a higher RP level than tg

**Comment by KingPhilipIII** [comment]
I wouldn't necessarily say a higher level of roleplay, atleast not much, more really their pacing is far slower and much less chaotic than TG.

**Comment by Armhulen** [comment]
the conversion mode culling is coming, conversion modes can be good but too many of them is bad so we can just kill two other conversion modes for this!

**Comment by Armhulen** [comment]
see two for one, you get value on removing them and i get my good conversion mode that proves all of you wrong

**Comment by RemieRichards** [comment]
@Armhulen just accept it's not happening

**Comment by CPTANT** [comment]
Kor has already outlined the reasons why shadowling is a bad gamemode.

And playing the sling itself isn't even fun. All you do is sit in a room in maint waiting ages to convert people.

**Comment by Armhulen** [comment]
That seems like a bad shadowling player, then.

**Comment by MoreRobustThanYou** [comment]
How to actually play slings: actively hunt for people, smash lights, inch into the hallway / medbay / whatever

**Comment by RemieRichards** [comment]
@tgstation/commit-access 
Calling a Vote: Y/N to shadowling (only for maintainers if you can't read the @ about)

**Comment by Cyberboss** [comment]
Y

**Comment by AnturK** [comment]
I'm fine with the theme, but if this is just reheat of the old mode this gets a no from me.

**Comment by ChangelingRain** [comment]
N

**Comment by MoreRobustThanYou** [comment]
@AnturK It's been given some changes (drain life removed, cool new sprites, yog shadowwalk, longer implant removal time, better ascendant sprites, ways to kill ascendants, thrall requirement scaling)

**Comment by lzimann** [comment]
N

**Comment by MoreRobustThanYou** [comment]
**Balance ideas**:
* Make thralls constantly hallucinate shadowy figures. This might make being a thrall worse, discouraging maint greytide as soon as someone screams "Slings"!
* Replace glare with an ability that lets a shadowling do considerable stamina damage to someone on a dark tile, but not enough to knock down.

Any other ideas? tell me

**Comment by Super3222** [comment]
Just finish umbrage instead.

**Comment by MoreRobustThanYou** [comment]
I have a feeling this is slowly going to turn into a version of umbrage...

**Comment by MoreRobustThanYou** [comment]
Ok.

Glare is no longer "hurr durr i stared at you and now i can cuff you", now it only causes stamina damage and stuttering, amount only based on distance. It still does the same as the old glare when you are _directly next to the sling_, but any further and it's ONLY stamina damage and stuttering.

**Comment by KingPhilipIII** [comment]
Be careful that you don't end up like Joan and irreparably damage an antagonist in your honest attempt to bring it to an acceptable level. 

Glare is to the shadowlings like geis is to the clockies.

**Comment by MoreRobustThanYou** [comment]
@KingPhilipIII and geis stuns at melee range. So does newglare.

**Comment by KingPhilipIII** [comment]
I don't mean that. My point is it's an integral part of the antagonist's gameplay. You have to be careful when you touch it.

**Comment by CPTANT** [comment]
I think the shadowling concept is only interesting as a side antag and not as a conversion mode.

**Comment by optimumtact** [comment]
it's a no lol

**Comment by MoreRobustThanYou** [comment]
better than gangs at least

**Comment by Armhulen** [comment]
booooooooooooooooooooo

**Comment by Armhulen** [comment]
orrrrrrrrrrranges you take this back right now!

**Comment by MoreRobustThanYou** [comment]
what arm said

**Comment by KorPhaeron** [comment]
Tommorow night I will help start shadowling as a side antag please be calm

**Comment by kilkun** [comment]
well this is gay

**Pull request closed without merging**
